sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -95,7 +97,8 @@ class Iso8601TimestampFormatter(
  * @param zoneId the time zone identifier in which the formatter parses or format timestamps
  */
 class FractionTimestampFormatter(zoneId: ZoneId)
```

Comment:
Seems we can remove it, and simply create `Iso8601TimestampFormatter` with `isParsing =true`.

Reply:
do you mean false here, this formatter is used in formating not parsing as I see in the caller side

Reply:
do we do it in this or another pr?

Reply:
Now I'm wondering if `isParsing` is a good name. Here we do formatting, but we want to use var-length second fraction to avoid padding `0` at the end.

How about renaming to `needVarLengthSecondFraction`? We can use `false` as the default value then many tests do not need to be changed.

Reply:
BTW I'm OK to remove `FractionTimestampFormatter` in followup.

Reply:
OK, I will check that after this one